test/functional/inspec_exec_junit_test.rb

```diff
@@ -11,55 +11,55 @@
     out = inspec('exec ' + example_control + ' --format junit --no-create-lockfile')
     out.stderr.must_equal ''
     out.exit_status.must_equal 0
-    doc = Nokogiri::XML(out.stdout)
-    doc.errors.length.must_equal 0
+    doc = REXML::Document.new(out.stdout)
+    doc.has_elements?.must_equal true
   end
 
   it 'can execute the profile with the junit formatter' do
     out = inspec('exec ' + example_profile + ' --format junit --no-create-lockfile')
     out.stderr.must_equal ''
     out.exit_status.must_equal 0
-    doc = Nokogiri::XML(out.stdout)
-    doc.errors.length.must_equal 0
+    doc = REXML::Document.new(out.stdout)
+    doc.has_elements?.must_equal true
```

Comment:
No. Nokogiri will happily try and parse anything you throw at it and return a `Nokogiri::XML::Document` object. It will not raise an exception unless you enable strict parsing.

So the way to determine if Nokogiri parsed the XML successfully is to call `#errors` on it, or set string parsing and catch the exception.

REXML is somewhat similar... it will take any string or `IO` object and try to return a `REXML::Document` object. You can tell if it's valid XML if it has any `REXML::Element` objects contained within it.